### PR TITLE
Fix errors due to singular and plural conversion in metaserver

### LINKED ISSUE #5890: Get a DeviceList through the API provided by MetaServer on the edge. The Kind field in the return information is incorrect.
**What happened**:

When I get `*v1beta1.DeviceList` using the `clientSet` provided by kubeedge:
```go
	deviceList, err := versionedClient.DevicesV1beta1().Devices(deviceRequest.Namespace).List(ctx, metaV1.ListOptions{
		LabelSelector: deviceRequest.LabelSelector,
	})
	if err != nil {
		return nil, fmt.Errorf("failed to list devices: %v", err)
	}
```
an error occurs:
error

Then I call MetaServer API directly: 
```shell
curl 'http://127.0.0.1:10550/apis/devices.kubeedge.io/v1beta1/namespaces/default/devices'
```
The Kind field in the return information is `DevicList` :
image

**What you expected to happen**:

The Kind field in the return information is `DeviceList`.

**Anything else we need to know?**:

This bug might be related to suffix handling.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.28.14
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`): v1.18.0
(I also found this bug in kubeedge1.17)

### PR DESCRIPTION
**Pull request by wbc6080**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

As mentioned in the #5890 and #4471 , metaserver does not accurately convert singular and plural type when parsing resource types. It converts `devices` into `devic` and `namesapces` into `namespac`, so the resource type obtained is incorrect. So we need to modify the method of singular and plural conversion.

**Which issue(s) this PR fixes**:

Related to #5890 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by wbc6080 — COMMENTED** [review]
Since the rules for converting English words between singular and plural are complicated, we use the `github.com/gertd/go-pluralize `package for conversion. I tested the performance of this package, and the results are as follows. We can see that this package can more accurately convert various types of words between singular and plural.
image

File: pkg/metaserver/util/util.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"strings"
 
+	pluralize "github.com/gertd/go-pluralize"
```
Comment (fisherxu) [inline pkg/metaserver/util/util.go:7]: Looks good, but it's better to keep same with Kubernetes. Does K8s also use this pkg?

Comment (fisherxu) [reply pkg/metaserver/util/util.go:7]: It's better to keep same with K8s.

Comment (fujitatomoya) [reply pkg/metaserver/util/util.go:7]: @wbc6080 thanks for taking care of this. this is tricky but annoying problem.

as @fisherxu mentions, probably we would want to go with k8s approach for now. but your concern also stands, we would want keep the note for future possible problems though.

Comment (wbc6080) [reply pkg/metaserver/util/util.go]: I have made modifications based on the processing method in k8s, and also retained the original issue as a reference.

File: pkg/metaserver/util/util.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func UnsafeResourceToKind(r string) string {
 	switch {
 	case strings.HasSuffix(k, "ies"):
 		return strings.TrimSuffix(k, "ies") + "y"
-	case strings.HasSuffix(k, "es"):
+	case strings.HasSuffix(k, "ses"):
```
Comment (fisherxu) [inline pkg/metaserver/util/util.go:69]: Is it need for case `es`?

Comment (wbc6080) [reply pkg/metaserver/util/util.go:69]: In the implementation of k8s, `es` is added only when the word ends with `s`. https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/restmapper.go#L142

**Comment by wbc6080** [comment]
PTAL, thanks
cc @WillardHu @Shelley-BaoYue @fisherxu

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**